File: blog-service/src/main/java/net/blueshell/blogservice/repository/BlogRepository.java
package net.blueshell.blogservice.repository;

import jakarta.validation.constraints.NotNull;
import net.blueshell.blogservice.model.Blog;
import net.blueshell.db.BaseRepository;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface BlogRepository extends BaseRepository<Blog, UUID> {

    @Query("SELECT n FROM Blog n ORDER BY n.publishedAt DESC")
    @Override
    Page<Blog> findAll(@NotNull Pageable pageable);

    @Query("SELECT n FROM Blog n ORDER BY n.publishedAt DESC")
    @Override
    List<Blog> findAll();
}

File: api/src/main/kotlin/net/blueshell/api/repository/BlogRepository.kt
package net.blueshell.api.repository

import jakarta.validation.constraints.NotNull
import net.blueshell.api.base.BaseRepository
import net.blueshell.api.model.Blog
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface BlogRepository : BaseRepository<Blog?> {
    @Query("SELECT n FROM Blog n ORDER BY n.publishedAt DESC")
    override fun findAll(@NotNull pageable: @NotNull Pageable): Page<Blog?>

    @Query("SELECT n FROM Blog n ORDER BY n.publishedAt DESC")
    override fun findAll(): MutableList<Blog?>
}
